analysis/priestholy/src/modules/shadowlands/items/FlashConcentration.tsx

```diff
@@ -0,0 +1,228 @@
+      event.timestamp -
+      (event.channel?.beginChannel?.timestamp
+        ? event.channel.beginChannel.timestamp
+        : event.timestamp);
+  }
+
+  statistic() {
+    // HealHPSNoFC gives the HPS which would have been obtained by casting the same number of heals without Flash Concentration Buff
+    const HealHPSNoFC =
+      (this._totalHeal - this._gainedHeal) / (this._totalCastTime + this._gainedCastTime);
+    // BuffedHealHPS gives the total HPS of Heal in this fight
+    const BuffedHealHPS = this._totalHeal / this._totalCastTime;
+    return (
+      <Statistic
+        size="flexible"
+        category={STATISTIC_CATEGORY.ITEMS}
+        tooltip={
+          <>
+            Impact is approximative since <SpellLink id={SPELLS.FLASH_CONCENTRATION.id} /> modifies
+            the gameplay in depth
+          </>
+        }
+      >
+        <BoringSpellValueText spell={SPELLS.FLASH_CONCENTRATION}>
+          <ItemHealingDone
+            // approximate considering the player would have cast heal without FC
+            amount={(BuffedHealHPS - HealHPSNoFC) * this._totalCastTime}
+            approximate
+          />
+          <br />
+          {this._gainedCastTime / 1000} sec saved on <SpellLink id={SPELLS.GREATER_HEAL.id} /> casts
+        </BoringSpellValueText>
+      </Statistic>
+    );
+  }
+
+  get dropSuggestion() {
+    return {
+      actual: this._totalDrops,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 0,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  get wastefulFlashHealSuggestion() {
+    return {
+      actual: this._wastefulCasts,
+      isGreaterThan: {
+        minor: 0,
+        average: (2 * this.owner.fightDuration) / 1000 / 60,
+        major: (4 * this.owner.fightDuration) / 1000 / 60,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  suggestions(when: When) {
+    when(this.dropSuggestion).addSuggestion((suggest, actual) =>
+      suggest(
+        <span>
+          You dropped your <SpellLink id={SPELLS.FLASH_CONCENTRATION.id}></SpellLink> stacks{' '}
+          {actual} times. Try to keep it up at all times.
+        </span>,
+      )
+        .icon(SPELLS.FLASH_CONCENTRATION.icon)
+        .staticImportance(SUGGESTION_IMPORTANCE.MAJOR)
+        .actual(<>You lost your stacks {actual} times</>)
+        .recommended(<>It is recommended to maintain 5 stacks at all times</>),
+    );
+
+    when(this.wastefulFlashHealSuggestion).addSuggestion((suggest, actual) =>
+      suggest(
+        <span>
+          You used too many unnecessary <SpellLink id={SPELLS.FLASH_HEAL.id} /> while you had 5
+          stacks of <SpellLink id={SPELLS.FLASH_CONCENTRATION.id}></SpellLink>
+        </span>,
+      )
+        .icon(SPELLS.FLASH_CONCENTRATION.icon)
+        .actual(
+          <>
+            You cast {actual} times while there were more than 5 seconds left of{' '}
+            <SpellLink id={SPELLS.FLASH_CONCENTRATION.id} />,
+          </>,
```

Comment:
So it's always bad to cast Flash Heal at max stacks? Is this because at max stacks Heal is a strictly better spell? If so, you should explicitly say that in the suggestion.

Reply:
So, no, but this is a good enough approximation would be my 2-cents (with a little tweak, see below). There are two cases when it is ok to cast Flash Heal before the buff is about to drop:
- When you have 2 stacks of surge of light (which can def happen, truthfully I forgot about this case, but would fit if I made the minor threshold at 1 per minute, I think, alternatively, we could check for it, although, it is more complex)
- When someone will die in between the time Flash Heal would have healed and the time Heal heals, which is a very very narrow window at 5 stacks, so I chose to ignore that, because it is pretty complex and not very impactful imo.
That's another case where a more experienced point of view would help for sure

Reply:
>     * When you have 2 stacks of surge of light (which can def happen, truthfully I forgot about this case, but would fit if I made the minor threshold at 1 per minute, I think, alternatively, we could check for it, although, it is more complex)
It's very easy to check, try `this.selectedCombatant.getBuffStacks(SPELLS.SURGE_OF_LIGHT_BUFF.id) >= 2`

Reply:
oh yes indeed, thanks!